dev/doc/style.txt

```diff
@@ -67,6 +67,14 @@ OCaml Style:
     (setq tuareg-function-indent 0)
     (setq tuareg-let-always-indent nil)
 
+  - when a match triggers warning 40 (eg "match x with VarRef x -> bla
+    | ConstRef x -> bli | _ -> blo" when GlobRef is not open) it can be
+    resolved in several ways:
+    + locally open GlobRef
+    + type annotate "x : GlobRef.t" (where it is introduced, or in the match expression, whichever is nicer)
+    + annotate the first branch "GlobRef.VarRef x -> bla"
+      this last solution is not robust to branch reordering so should not be prefered
```

Comment:
This piece of doc is not super clear: from the discussion, it seems like option 2 and 3 are acceptable options, but they do not actually make the warning go away. This is fine, but the doc makes the reader think that they do. Another disadvantage of referring to the warning in this piece of doc is that in practice, since we disable the warning with Dune, I'm expecting that developers won't ever see the warning in practice.

Reply:
Yes, needs to be amended. This is actually not "when warning 40 is triggered" but when Coq errors which "can't resolve the constructor" I think.

Reply:
right